### ISSUE TITLE
(Enhancement) Fit to diagram should include text and connections

### ISSUE THREAD

**Issue by ceraolo** [OpenModelica/OpenModelica#11723]
### Description
When clicking on "Fit to diagram", the model coordinate system is adapted to that to include the whole diagram. Currently, it does not include text around models e.g. the text containing the instance name.
Moreover, not always it includes connection lines.
I propose to enhance the algorithm connected to "Fit to diagram" so that also this text and connections are always enclosed. 

### Steps to Reproduce
Click "Fit to Diagram" on the following model:
```
model M1
  Modelica.Blocks.Continuous.FirstOrder firstOrder annotation(
    Placement(transformation(origin = {2, 90}, extent = {{-10, -10}, {10, 10}})));
Modelica.Blocks.Sources.Step step annotation(
    Placement(transformation(origin = {-48, 90}, extent = {{-10, -10}, {10, 10}})));
equation
connect(firstOrder.u, step.y) annotation(
    Line(points = {{-10, 90}, {-36, 90}}, color = {0, 0, 127}));
  connect(firstOrder.y, firstOrder.u) annotation(
    Line(points = {{14, 90}, {24, 90}, {24, 72}, {-20, 72}, {-20, 90}, {-10, 90}}, color = {0, 0, 127}));

annotation(
    Diagram(coordinateSystem(extent = {{-60, 100}, {20, 60}})),
  version = "",
  uses);
end M1;

```

This is my screen after clicking:
![image](https://github.com/OpenModelica/OpenModelica/assets/7373030/aaa06d2c-7636-43fa-bbb8-dc15485fef35)

Second example.: 
```
model M1
  Modelica.Blocks.Continuous.FirstOrder firstOrder annotation(
    Placement(transformation(origin = {2, 90}, extent = {{-10, -10}, {10, 10}})));
Modelica.Blocks.Sources.Step step annotation(
    Placement(transformation(origin = {-48, 90}, extent = {{-10, -10}, {10, 10}})));
equation
connect(firstOrder.u, step.y) annotation(
    Line(points = {{-10, 90}, {-36, 90}}, color = {0, 0, 127}));

annotation(
    Diagram(coordinateSystem(extent = {{-60, 100}, {20, 80}})),
  version = "",
  uses);
end M1;

```

Result:

![image](https://github.com/OpenModelica/OpenModelica/assets/7373030/b551c205-cd5d-4d5a-8ab6-7b69dbb1e527)

OM 1.23.0-dev-193

**Comment by ceraolo** [comment]
I did close this because the enhancement proposal has so a little impact that does not deserve attention.

**Comment by casella** [comment]
Maybe we just need to leave some free boundary space anyway. I guess this would solve most of the problems.

**Comment by ceraolo** [comment]
I decided to close this ticket considering the rationale beyond the very useful key "fit to diagram" that I contributed to create.

_The first thing that I would like to say is that the presence of the white page in diagrams (due to the diagram extent command present in all models), IMO does noting useful, it even makes some small damage, but is part of in Modelica language and we must live with it._

"Fit to Diagram" (it should be "fit to extent") allows to see the diagram at the largest possible size, without having to zoom every time we open it. This is epecially useful using Diagram Window in plotting perspective in subwindow view, where screen space is occupied also by plot windows.
  
Yesterday, after having opened the ticket I realised that, if "Fit to Diagram" leaves the diagram slightly larger than the extent, it **makes not harm at all (it may even be beneficial!).**
That's why I closed it.
